### PR TITLE
AppManifest `version` becomes `installed_hash`

### PR DESCRIPTION
**Pull request by maackle**
### Summary

`version` is renamed to `installed_hash`, and now takes only a single DnaHashB64 rather than a list.

This will be used to look up existing DNAs upon app installation, to avoid the need to pass in DNAs multiple times when re-installing an app which has already been installed.

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain_types/src/app/app_bundle.rs [hunk 1]
```diff
@@ -166,29 +192,53 @@ impl AppBundle {
...
         clone_limit: u32,
         modifiers: DnaModifiersOpt,
     ) -> AppBundleResult<CellProvisioningOp> {
-        let bytes = self.resolve(location).await?;
-        let dna_bundle: DnaBundle = mr_bundle::Bundle::decode(&bytes)?.into();
-        let (dna_file, original_dna_hash) = dna_bundle.into_dna_file(modifiers).await?;
-        if let Some(spec) = version {
-            if !spec.matches(original_dna_hash) {
-                return Ok(CellProvisioningOp::NoMatch);
+        let dna_file = if let Some(hash) = installed_hash {
```
Comment (jost-s) [inline crates/holochain_types/src/app/app_bundle.rs:201]: This looks correct to me, but could use a test or two.

**Review by jost-s — APPROVED** [review]

**Comment by ThetaSinner** [comment]
How unique are DNA hashes expected to be? A limited number of required fields are included in the hash so it's relatively easy to colide with somebody else's app. It's not currently obvious if you do that with your own app https://github.com/holochain/holochain/issues/2145. 

Should this logic be tied to the exact code that is intended to run? It looks intentional that the code doesn't do that but I'm not clear about why that is

**Comment by neonphog** [comment]
> How unique are DNA hashes expected to be?

The wasm itself is included in the hash, so should be as unique as the zomes you are using. Is anyone building hApps by just including premade zomes?

**Comment by ThetaSinner** [comment]
Ah okay, I see what's going on there now - it's only the integrity zomes that get hashed, but not the coordinator zomes. So if somebody does something like use the scaffolding tool and only updates their coordinator zome but leaves the integrity zome at default then they risk getting a collission

**Review by ThetaSinner — APPROVED** [review]

**Pull request merged**